docs/src/pages/presets/writing-presets/index.md

```diff
@@ -0,0 +1,120 @@
+---
+id: 'writing-presets'
+title: 'Writing Presets'
+---
+
+[Storybook presets](../introduction/) are grouped collections of `babel`, `webpack`, and `addons` configurations that support specific use cases in Storybook, such as typescript or MDX support.
+
+This doc covers the [presets API](#presets-api) and how to use the presets mechanism for [advanced configuration](#advanced-configuration).
+
+## Presets API
+
+A preset is a set of hooks that is called by Storybook on initialization and can override configurations for `babel`, `webpack`, `addons`, and `entries`.
+
+Each configuration has a similar signature, accepting a base configuration object and options, as in this webpack example:
+
+```js
+export async function webpack(baseConfig, options) {
+  // modify or replace config
+  config.module.rules.push(/* some loader */);
+  return config;
+}
+```
+
+### Babel
+
+The babel functions `babel`, `babelDefault`, and `managerBabel` all configure babel in different ways.
+
+All functions take a [Babel configuration object](https://babeljs.io/docs/en/configuration) as their argument and can modify it or return a new object.
+
+For example, Storybook's Vue support uses presets internally and here's how it configures babel:
+
+```js
+export function babelDefault(config) {
+  return {
+    ...config,
+    presets: [...config.presets, require.resolve('babel-preset-vue')],
+  };
+}
+```
+
+- `babelDefault` is applied to the preview config before any
+- `babel` is applied to the preview config, after it has been
+- `managerBabel` is applied to the manager.
+
+### Webpack
+
+The webpack functions `webpack`, `webpackFinal`, and `managerWebpack` configure webpack.
+
+All functions take a [webpack4 configuration object](https://webpack.js.org/configuration/).
+
+For example, here is how Storybook automatically adopts `create-react-app`'s configuration if it's installed, where `applyCRAWebpackConfig` is a set of smart heuristics for modifying the input config.
+
+```js
+export function webpackFinal(config, { configDir }) {
+  if (!isReactScriptsInstalled()) {
+    logger.info('=> Using base config because react-scripts is not installed.');
+    return config;
+  }
+
+  logger.info('=> Loading create-react-app config.');
+  return applyCRAWebpackConfig(config, configDir);
+}
+```
+
+- `webpack` is applied to the preview config after
+- `webpackFinal` is applied to the preview config after
```

Comment:
I think this needs filling out

Reply:
👍 updated, could use @igor-dv 's comments here